azure-functions/src/durable.rs

```diff
@@ -59,30 +60,56 @@ unsafe fn waker_wake_by_ref(_: *const ()) {
 
 unsafe fn waker_drop(_: *const ()) {}
 
-/// The entrypoint for orchestration functions.
-///
-/// The given future is the user function.
+fn poll_function<T>(func: impl Future<Output = T>) -> Poll<T> {
+    let waker = unsafe {
+        Waker::from_raw(RawWaker::new(
+            null(),
+            &RawWakerVTable::new(waker_clone, waker_wake, waker_wake_by_ref, waker_drop),
+        ))
+    };
+
+    Future::poll(Box::pin(func).as_mut(), &mut Context::from_waker(&waker))
+}
+
+/// The entrypoint for orchestration functions that do not output a value.
 #[doc(hidden)]
 pub fn orchestrate(
     id: String,
     func: impl Future<Output = ()>,
     result: Rc<RefCell<ExecutionResult>>,
 ) -> InvocationResponse {
-    let waker = unsafe {
-        Waker::from_raw(RawWaker::new(
-            null(),
-            &RawWakerVTable::new(waker_clone, waker_wake, waker_wake_by_ref, waker_drop),
-        ))
+    match poll_function(func) {
+        Poll::Ready(_) => result.borrow_mut().done = true,
+        Poll::Pending => {}
     };
 
-    match Future::poll(Box::pin(func).as_mut(), &mut Context::from_waker(&waker)) {
-        Poll::Ready(_) => {
-            // Orchestration has completed and the result is ready, return done with output
-            result.borrow_mut().done = true;
-        }
-        Poll::Pending => {
-            // Orchestration has not yet completed
+    InvocationResponse {
```

Comment:
Let me extract this into a method so both `orchestrate` functions can return the exact same thing.